### PR TITLE
Closes #738: Automatically detect new releases

### LINKED ISSUE #738: Automatically check for new releases
NetBox should periodically check for a new release and inform the user if one exists. PR #737 was submitted to introduce this feature, but I'd like to see a more efficient approach. This will require implementing a database model to store temporary configuration information. We'll also need to introduce a permanent configuration variable which allows a user to disable this feature.

### PR DESCRIPTION
**Pull request by jeremystretch**
### Fixes: #738

(Successor to #4001)

- Establishes a new Redis queue for release checking jobs
- Renames the `webhooks` Redis connection to `tasks` (backward-compatible until v2.9)
- Introduces the `UPDATE_REPO_URL` and `UPDATE_CACHE_TIMEOUT` config parameters
- Checks for latest release once per timeout interval and caches result

### CODE REVIEW

**Comment by jeremystretch** [comment]
We need to figure out how to avoid enqueuing redundant jobs. Currently, if the rqworker process is not running, NetBox will continually add new jobs to check for new releases. There is never a need to have more than one such job in the queue.

**Comment by jeremystretch** [comment]
This should be about ready to merge. Just looking for one or two people to review. I'll take a fresh look at it again later today or tomorrow.

**Review by lampwins — COMMENTED** [review]
This looks good to me, I have just a couple of general comments, mostly around logging.

File: netbox/utilities/releases.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+from django_rq import get_queue
+
+from utilities.background_tasks import get_releases
+
+logger = logging.getLogger('netbox.releases')
+
+
+def get_latest_release(pre_releases=False):
+    if settings.UPDATE_REPO_URL:
+        logger.debug("Checking for most recent release")
```
[the remark this answers is not in the document]
Comment (jeremystretch) [reply netbox/utilities/releases.py]: This happens at every load of `home()` by an admin user though. I'll change the actual job queuing to info but I think the rest of the debugs are fine.

File: netbox/utilities/background_tasks.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            if 'tag_name' not in release:
+                continue
+            if not pre_releases and (release.get('devrelease') or release.get('prerelease')):
+                continue
+            releases.append((version.parse(release['tag_name']), release.get('html_url')))
+        logger.debug("Found {} releases; {} usable".format(total_releases, len(releases)))
+
+    except requests.exceptions.RequestException:
+        # The request failed. Set a flag in the cache to disable future checks to this URL for 15 minutes.
+        logger.exception("Error while fetching {}. Disabling checks for 15 minutes.".format(url))
```
[the remark this answers is not in the document]
Comment (jeremystretch) [reply netbox/utilities/background_tasks.py]: `.exception(msg)` is just a wrapper for `.error(msg, exc_info=True)`

File: docs/configuration/optional-settings.md [hunk 3]
```diff
@@ -351,6 +351,24 @@ The time zone NetBox will use when dealing with dates and times. It is recommend
 
 ---
 
+## UPDATE_CACHE_TIMEOUT
```
[the remark this answers is not in the document]
Comment (jeremystretch) [reply docs/configuration/optional-settings.md]: Renaming these settings to `RELEASE_CHECK_TIMEOUT` and `RELEASE_CHECK_URL`

File: netbox/utilities/releases.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        try:
+            latest_release = cache.get('latest_release')
+            if latest_release:
+                logger.debug("Found cached release: {}".format(latest_release))
+                return latest_release
```
[the remark this answers is not in the document]
Comment (jeremystretch) [reply netbox/utilities/releases.py]: It does; the whole `(version, URL)` tuple is cached as "latest_release".

**Pull request merged**